File: 002/euler2.pl
sub f {
  my $sum = 0;
  my ($a, $b) = (0, 1);
  while ($b <= $_[0]) {
    $sum += $b if $b % 2 == 0;
    ($a, $b) = ($b, $a + $b);
  }
  $sum;
}


print f(4000000) . "\n";

File: 002/euler2.js
module.exports = function(n) {
  var sum = 0;
  var a = 0, b = 1;
  while (b <= n) {
    if (b % 2 == 0) {
      sum += b;
    }
    b = a + b;
    a = b - a;
  }
  return sum;
};


if (process.argv[1] == __filename) {
  console.log(module.exports(4000000));
}
